src/auto/AutoModule.lua

```diff
@@ -0,0 +1,46 @@
+local AutoModule, parent = torch.class('nn.AutoModule', 'nn.Module')
+
+-- Constructor needs a function and params (if any)
+-- The input function has the format (for Linear + ReLU):
+-- function(params, input)
+--    local y = params.weight * input + params.bias
+--    local output = torch.mul( torch.abs( y ) + y, 0.5)
+--    return output
+-- end
+function AutoModule:__init(fn, weight, bias)
+   parent.__init(self)
+
+   self.d = require 'autograd'
```

Comment:
Typically not a great idea to include a package here, as it will get serialized when serializing the object to disk for instance. Autograd is pure Lua now so it's ok, but if it had C functions it would fail.